infra/provision.go

```diff
@@ -0,0 +1,303 @@
+package main
+
+import (
+	_ "embed"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/pulumi/pulumi-command/sdk/go/command/remote"
+	"github.com/pulumi/pulumi-gcp/sdk/v6/go/gcp/compute"
+	"github.com/pulumi/pulumi-gcp/sdk/v6/go/gcp/projects"
+	"github.com/pulumi/pulumi-gcp/sdk/v6/go/gcp/serviceaccount"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
+	"github.com/umee-network/umee-infra/infra/pulumi/common/resources"
+
+	"contracts/unit"
+)
+
+func (network Network) Provision(ctx *pulumi.Context, secrets []NodeSecretConfig) error {
+	var addrs pulumi.StringArray
+	var nodeHostNames []string
+
+	moniker := genMoniker(network.ChainID)
+	location := network.NodeConfig.Location
+	addr, err := compute.NewAddress(ctx, moniker+"-ip", &compute.AddressArgs{
+		Labels: pulumi.StringMap{
+			"chain_id": pulumi.String(network.ChainID),
+		},
+		NetworkTier: pulumi.String("STANDARD"),
+		Region:      pulumi.String(location.Region),
+	})
+	if err != nil {
+		return err
+	}
+
+	addrs = append(addrs, addr.Address)
+
+	nodeHostName := fmt.Sprintf("%s.%s.node.ojo.network", moniker, network.ChainID)
+	nodeHostNames = append(nodeHostNames, nodeHostName)
+
+	bootDisk := &compute.InstanceBootDiskArgs{
+		DeviceName: pulumi.String(fmt.Sprintf("%s-bootdisk", moniker)),
+		InitializeParams: &compute.InstanceBootDiskInitializeParamsArgs{
+			Image: pulumi.String("family/ubuntu-minimal-2204-lts"),
+			Type:  pulumi.String(network.NodeConfig.DiskType),
+			Size:  pulumi.Int(network.NodeConfig.DiskSizeGB),
+		},
```

Comment:
Are we spinning up a node here? I thought the goal should be to spin up a relayer to an existing contract quickly